frontend/components/songs/song_card_dropdown.jsx

```diff
@@ -3,123 +3,130 @@ import React, { useState, useEffect, useRef } from "react";
 import SongCardSubmenu from "./song_card_submenu";
 import SongCardDropdownItem from "./song_card_dropdown_item";
 
-const SongCardDropdown = ({
-    currentItem,
-    playlists,
-    currentUser,
-    history,
-    selectedSong,
-    updateSongCardDropdownState,
-    items,
-    depthLevel,
-    dropdownPosition,
-    removePlaylisted,
-    createNewPlaylisted,
-    createPlaylist,
-    displayPlaylist,
-}) => {
-    // Set local state for SongCardSubmenu
-    let dropdownRef = useRef();
-    const [submenuState, setSubmenuState] = useState({ isOpen: false });
+const SongCardDropdown = forwardRef(
+    (
+        {
+            currentItem,
+            playlists,
+            currentUser,
+            history,
+            selectedSong,
+            songCardDropdownState,
+            updateSongCardDropdownState,
+            items,
+            depthLevel,
+            dropdownPosition,
+            removePlaylisted,
+            createNewPlaylisted,
+            createPlaylist,
+            displayPlaylist,
+        },
+        ref
+    ) => {
+        // Set local state for SongCardSubmenu
 
-    // Add event listeners when menu is open; remove when menu is closed
-    useEffect(() => {
-        const handler = (event) => {
-            if (
-                submenuState.isOpen &&
-                dropdownRef.current &&
-                !dropdownRef.current.contains(event.target)
-            ) {
-                updateSongCardDropdownState({ isOpen: false });
-                setSubmenuState({ isOpen: false });
-            }
-        };
-        document.addEventListener("mousedown", handler);
-        document.addEventListener("touchstart", handler);
-        return () => {
-            // Cleanup the event listener when component unmounts
-            document.removeEventListener("mousedown", handler);
-            document.removeEventListener("touchstart", handler);
-        };
-    }, [submenuState]);
+        const [submenuState, setSubmenuState] = useState({ isOpen: false });
 
-    const toggleSubmenuAndPlaceDropdown = (e) => {
-        e.preventDefault();
-        setSubmenuState({ isOpen: !submenuState.isOpen });
-    };
+        // Add event listeners when menu is open; remove when menu is closed
+        useEffect(() => {
+            const whenMenuIsOpen = (event) => {
+                if (
+                    songCardDropdownState.isOpen &&
```

Comment:
Seems like there's some logic changes, was this intended?

Reply:
Sorry, had this PR as depends on #66. Will merge #66 and repush.

Reply:
Did I do this correctly? This branched off the last PR, but PR #66 wasn't approved yet. I did git rebase -i and took off the commit concerning PR #66 so you could see the different from main, and I noted depends in the PR description.

Reply:
Reverting because it seems I did not do this correctly and the dropdownRef was lost after this PR.